File: backend_py/app/routers/runs.py
"""Run details & results."""

from fastapi import APIRouter, Query

from app.deps import DBMain
from app.services.testmo import testmo_service

router = APIRouter()


@router.get("/{run_id}")
async def get_run(run_id: int, db: DBMain):
    run = await testmo_service.get_run_details(run_id)
    return run


@router.get("/{run_id}/results")
async def get_run_results(run_id: int, status: str | None = Query(None), db: DBMain = None):
    results = await testmo_service.get_run_results(run_id, status_filter=status)
    return {"results": results}

File: frontend/src/server/trpc/routers/runs.ts
/**
 * ================================================
 * tRPC ROUTER — Runs
 * ================================================
 */

import { z } from 'zod';
import { router, publicProcedure } from '../init';
import testmoService from '../../services/testmo.service';

const runIdInput = z.object({
  runId: z.number().int().positive(),
});

const runResultsInput = z.object({
  runId: z.number().int().positive(),
  status: z.string().optional(),
});

export const runsRouter = router({
  details: publicProcedure
    .input(runIdInput)
    .query(async ({ input }) => {
      const runDetails = await testmoService.getRunDetails(input.runId);
      return { success: true as const, data: runDetails, timestamp: new Date().toISOString() };
    }),

  results: publicProcedure
    .input(runResultsInput)
    .query(async ({ input }) => {
      const results = await testmoService.getRunResults(input.runId, input.status || undefined);
      return { success: true as const, data: results, timestamp: new Date().toISOString() };
    }),
});
